5calls/app/src/main/java/org/a5calls/android/a5calls/model/AccountManager.java

```diff
@@ -0,0 +1,94 @@
+package org.a5calls.android.a5calls.model;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import org.a5calls.android.a5calls.FiveCallsApplication;
+
+/**
+ * A singleton responsible for managing all local account-related information
+ */
+public enum AccountManager {
+    Instance;
```

Comment:
I think it should be separate, since there's a clear separation of concerns here.  The account manager is just for managing account related, local information, while the AppSingleton is more of a stand in for the Application class.  Also, enums as singletons are pretty safe.